Cql/Cql.Compiler/CqlOperatorsBinding.cs

```diff
@@ -509,7 +509,7 @@ private Expression SortBy(MemberExpression operators, Expression source, Express
         {
             if (by is LambdaExpression lambda && order is ConstantExpression orderConstant && orderConstant.Type == typeof(ListSortDirection))
             {
-                var elementType = TypeResolver.GetListElementType(source.Type) ?? throw new InvalidOperationException($"{source.Type} was expected to be a list type.");
+                var elementType = TypeResolver.GetListElementType(source.Type).NotNull($"'{source.Type}' was expected to be a list type.");
```

Comment:
This is a meaningless change that reduces performance since method calls are more expensive than null coalescing.  Why do this?

Reply:
If it is inline as in the screenshot example below, then there is no expensive method call. I reverted the code in any case, so there is no more Check class

![image](https://github.com/FirelyTeam/firely-cql-sdk/assets/4639799/0ddc6e27-bb47-4dce-9cfe-f7764e2b0dbf)

Reply:
I actually like it, and I don't think these small performance differences mean anything in the compiling stage - reading the files, I/O, generating dynamic types, compiling Linq Expressions and calling the Roslyn compiler make these (inlined?) calls irrelevant.